import "union.proto";
import "resource.proto";

package SanProto;


//查看联盟捐献信息
message QueryUnionDonateInfoReq{
    required int32 union_id=1;
}
message QueryUnionDonateInfoRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    
    repeated UnionDonateTreasureBoxInfo boxes_info=3;
    optional UnionDonateInfo info=4;
}
//发起捐献
message OpenDonateReq{
    required int32 union_id=1;
    required int32 treasurebox_id =2;
}
message OpenDonateRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    optional UnionDonateTreasureBoxInfo box_info=3;

}
//进行捐献
message StartDonateReq{
    required int32 union_id=1;
    required int32 treasurebox_id=2;
    required DonateType donate_type=3;
    optional int32 money=4;
    optional int32 gold=5;
}
message StartDonateRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    optional UnionDonateInfo info=3;
	optional ResourceInfo resource=4;
	optional UnionDonateTreasureBoxInfo box_info=5;
}
//捐献进度请求
message DonateProgressReq{
 	required int32 union_id=1;
}
message DonateProgressRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    repeated UnionDonateTreasureBoxInfo boxes_info =3; 
}
//捐献记录请求
message DonateRecordReq{
    required int32 union_id=1;
}
message DonateRecoreRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    repeated bytes donate_record=3;
}
//领取宝箱奖励
message GetBoxRewardReq{
    required int32 union_id=1;
    required int32 treasurebox_id =2;
}
message GetBoxRewardRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    optional TreasureBoxReward reward =3;
}
//刷新宝箱
message RefreshBoxInfoReq{
    required int32 union_id=1;
    required int32 treasurebox_id =2;
}
message RefreshBoxInfoRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
    optional UnionDonateTreasureBoxInfo box_info=3;
}

//清空捐献冷却时间
message ClearDonateColdTimeReq{
    required int32 union_id=1;
	required int32 gold=2;
}
message ClearDonateColdTimeRes{
    required int32 status = 1;
    optional UNION_RET ret = 2;
	optional int32 cold_time=3;
	optional ResourceInfo resource=4;
}

